### PR TITLE
Build out snowflake_grants.py test suite for generate_schema_grants

### PR DESCRIPTION
**Pull request by blake-enyart**
Relatively small PR that:
* Brings a wide range of potential edge cases for the schema grants generation
* Minor refactors on some of the smaller if statements in snowflake_grants.py
* Refactor of _generate_revoke_select_privs to cut lines of code down by about a third

### CODE REVIEW

File: src/permifrost/core/permissions/utils/snowflake_grants.py [hunk 1]
```diff
@@ -1335,145 +1327,137 @@ def _generate_table_write_grants(self, conn, tables, shared_dbs, role):  # noqa
...
-        # SELECT is the only privilege for views so this covers both the read
-        # and write case since we have "all_grant_views" defined.
-        for granted_view in list(
-            set(self.grants_to_role.get(role, {}).get("select", {}).get("view", []))
-        ):
-            view_split = granted_view.split(".")
-            database_name = view_split[0]
+        for granted_resource in granted_resources:
+            resource_split = granted_resource.split(".")
+            database_name = resource_split[0]
```
Comment (PedramNavid) [inline src/permifrost/core/permissions/utils/snowflake_grants.py:1366]: for clarity could we name all three splits? not sure if `schema_name` and `table_name` conflict later on, but anything other than `resource_split[1]` has got to be easier to grok

**Review by PedramNavid — APPROVED** [review]
This is really nice, loving the refactor.

File: src/permifrost/core/permissions/utils/snowflake_grants.py (hunk 1 above)
Comment (blake-enyart) [reply src/permifrost/core/permissions/utils/snowflake_grants.py:1366]: Definitely doable. The more I'm looking into this revoke function, the more I'm realizing it is not really tested in most of the existing test harness. Going to work to come up with a test suite for that which I'll put into this PR as well just to make sure I'm not breaking anything existing

File: CHANGELOG.md [hunk 2]
```diff
@@ -126,7 +126,7 @@ Future grants on tables and views are granted on the database level.
 
 ### New
 * [#28](https://gitlab.com/gitlab-data/permifrost/-/issues/28) Add owner field
-  to all objects and the `require-owner` property
```
Comment (PedramNavid) [inline CHANGELOG.md:129]: This would end up being a breaking change, so I don't know if we should do this just yet. I set the entity to return `require_owner` but it reads from `require-owner` in the spec.
